include/fixed_containers/fixed_vector.hpp

```diff
@@ -1005,4 +1005,24 @@ constexpr typename FixedVector<T, MAXIMUM_SIZE, CheckingType>::size_type erase_i
     return original_size - c.size();
 }
 
+/**
+ * Construct a FixedVector with its capacity being deduced from the number of items being passed.
+ */
+template <typename T,
+          std::size_t MAXIMUM_SIZE,
+          fixed_vector_customize::FixedVectorChecking CheckingType =
+              fixed_vector_customize::AbortChecking<T, MAXIMUM_SIZE>>
+[[nodiscard]] constexpr FixedVector<T, MAXIMUM_SIZE, CheckingType> make_fixed_vector(
+    const T (&list)[MAXIMUM_SIZE],
+    const std_transition::source_location& loc =
+        std_transition::source_location::current()) noexcept
```

Comment:
If users need to customize the `CheckingType`, they would have to go through `MAXIMUM_SIZE` which defeats the size auto-deduction. To solve that, we could have a generic overload that has `MAXIMUM_SIZE` last:

```C++
template <typename T,
          fixed_vector_customize::FixedVectorChecking CheckingType,
          std::size_t MAXIMUM_SIZE>
[[nodiscard]] constexpr FixedVector<T, MAXIMUM_SIZE, CheckingType> make_fixed_vector(
    const T (&list)[MAXIMUM_SIZE],
    const CheckingType&,
    const std_transition::source_location& loc =
        std_transition::source_location::current()) noexcept
{
    FixedVector<T, MAXIMUM_SIZE, CheckingType> vector;
    for (decltype(MAXIMUM_SIZE) i = 0; i < MAXIMUM_SIZE; ++i)
    {
        vector.push_back(list[i], loc);
    }
    return vector;
}
```

and then use convenience overloads for the `CheckingType`. For the default one:

```C++
template <typename T, std::size_t MAXIMUM_SIZE>
[[nodiscard]] constexpr FixedVector<T,
                                    MAXIMUM_SIZE,
                                    fixed_vector_customize::AbortChecking<T, MAXIMUM_SIZE>>
make_fixed_vector(const T (&list)[MAXIMUM_SIZE],
                  const std_transition::source_location& loc =
                      std_transition::source_location::current()) noexcept
{
    return make_fixed_vector(list, fixed_vector_customize::AbortChecking<T, MAXIMUM_SIZE>{}, loc);
}
```

Users can have overloads similar to the 2nd one for custom `CheckingType`s . Having customization for `CheckingType` is a common scenario. For example, if a codebase primarily uses exceptions, you can have aliases whose purpose is to set the `CheckingType`, which for `FixedVector` would look like:
```C++
template<class T, std::size_t MAXIMUM_SIZE>
using FixedVector = fixed_containers::FixedVector<T, MAXIMUM_SIZE, ExceptionBasedChecking>;
```

Now, I realize that if a custom overload has to be provided anyway, then it is auto-deducing the size and can call the original function regardless, but by having the two overloads 
1) the code demonstrates how this situation is handled
2) It allows simple usage if the `CheckingType` doesn't depend on the other template parameters, for example: `        constexpr auto v1 = make_fixed_vector<int>({10, 11, 12, 13, 14}, MyAbortChecking{});`

Reply:
I did it:) Sorry for the delay, LMK if everything is good

Reply:
Looks good! I took the liberty of adding an additional template parameter on top of that, so if users have a child class (instead of a typedef) that does customization, it is possible to specify the return type of the `make_fixed_x()` functions.